capstone-app/src/app/modules/maps/map/map.component.ts

```diff
@@ -176,10 +176,24 @@ export class MapComponent implements OnInit, OnChanges {
   placesRequestFunc(location: google.maps.LatLng) {
     this.placesRequest.location = this.location;
     this.nearSearch.nearbySearch(this.placesRequest, results => {
+      console.log(results);
+      if (this.placesRequest.location !== this.location) {
+        console.log(this.placesRequest.location + ' doesnt equal ' + this.location);
+        return;
+      }
+      let city;
+      this.placeCitySharer.getCityName().subscribe(cities => {
+        city = cities;
+      })
+      if (city !== this.cityLocation) {
+        return;
```

Comment:
I'm a little skeptical about this code. The subscribe block will continue running after all this code is executed, meaning that an emission from this getCityName observable doesn't trigger this code to run. So on line 188 with city !== this.cityLocation, that city variable will not be updated with the "cities" result you're receiving from placesCitySharer. If you want to use the behavior subject it makes a lot more sense to call the function on the emission of the city so that the variables can get passed through and the code is ran with every emission. However, this leads to another problem. The only time city gets set is in this component file, so why do we need to call the the get function of the service rather than just declaring/checking a global city variable?

Reply:
This also looks like we have an async inside of an async

Reply:
Having the subscribe inside of the callback shouldn't be a problem. We can end the subscription after one run by including ".pipe(take(1))" before subscribe.

Setting city equal to cities inside the subscribe block won't work, because city won't be set outside of the subscribe block (It will run after the rest of this function).

Also, add semicolon on line 187.

Reply:
Would we even need this subscribe block though? Why not have a class variable, this.city, as that is equivalent to what's being returned from this service?

Reply:
The reason why we used this service method in other components was to share the city name across siblings, but that city name originated in this file and is never set again in any other file.

Reply:
+1